rust/catalyst-types/src/mmap_file.rs

```diff
@@ -0,0 +1,139 @@
+//! Memory-mapped file.
+
+use std::{
+    path::Path,
+    sync::{Arc, RwLock},
+};
+
+use fmmap::{MmapFile, MmapFileExt};
+use once_cell::sync::Lazy;
+use serde::Serialize;
+
+/// Memory-mapped file.
+pub struct MemoryMapFile {
+    /// The memory-mapped file.
+    file: MmapFile,
+    /// The size of the memory-mapped file.
+    size: u64,
+}
+
+/// Global statistic for memory-mapped files.
+static MEMMAP_FILE_STATS: Lazy<Arc<RwLock<MemMapFileStat>>> =
+    Lazy::new(|| Arc::new(RwLock::new(MemMapFileStat::default())));
+
+/// Memory-mapped file statistic.
+#[derive(Debug, Default, Clone, Serialize)]
+pub struct MemMapFileStat {
+    /// A counter for the number of memory-mapped files.
+    file_count: u64,
+    /// The total size of memory-mapped files.
+    total_size: u64,
+    /// The amount of time that memory-mapped files have been dropped.
+    drop_count: u64,
+    /// The total size of memory-mapped files that have been dropped.
+    drop_size: u64,
+    /// A count of errors encountered.
+    error_count: u64,
+}
+
+impl MemMapFileStat {
+    /// Get the statistic file count.
+    #[must_use]
```

Comment:
Currently there is no way to read the global MemMapFileStat holding the statistics.
Needs a `fn get() -> MemMapFileStat {}` function which locks the global stat for read, clones it, and then returns the cloned statistics.